src/docs/binding-functions.md

```diff
@@ -0,0 +1,103 @@
+
+Swift allows function homonyms and C# does not. In order to handle this, homonyms can be detected and the function name can have the name of the return type appended to it. This can get more complicated when the return types don't have pretty names on their own such as tuples or generic types
+or inner types.
+
+Like C#, Swift has keyword escaping, but uses backticks as delimeters instead of the at-sign prefix in C#.
+
+Operators are problematic as C# has only a limited set of available of operators whereas Swift has a very flexible set of operators.
+
+## ABI Differences
+The runtime team is taking care of both the self and return pointers as well as value type passing and returning.
+
+## Runtime Differences
+There are likely issues in mapping async functions.
+Generic functions in Swift work by adding an extra implicit argument for every generic in the function declaration which is a pointer to Swift type metadata for the generic argument. In addition, whatever argument is passed to the function gets passed as a reference. If a generic argument has one or more protocol constraints, the compiler injects an argument which is a pointer to the protocol witness table for the type with respect to the protocol. For example, if you have this Swift function:
+```swift
+public func Identity<T: SomeProtocol>(a: T) -> T {
+    return a
+}
+```
+When compiled, this function actual takes 3 arguments, not two:
+```swift
+public func Identity(a *T, type: T.Self, ProtoWitness(T.Self, SomeProtocol.Self)) -> T { }
+```
+
+## Idiomatic Differences
+Functions in swift start with a lowercase letter. We should auto-capitalize the first letter of a function name.
+Since property getters and setters are implemented as functions they map naturally onto C# properties. One difference, which is inconsequential except in projecting C# into Swift is that C# allows set-only properties whereas in Swift if a property has a setter it must have a getter.
+
+## Accessibility Differences
+For the most part, we should strive to address the language parity issues in ways that feel "right" in C# or at least "least bad". For example, this were definied in Swift:
+```swift
+prefix operator -+-
+
+public struct Container {
+    public var x: Int
+    public init (x: Int) {
+        self.x = x
+    }
+    public static prefix -+- (c: Container)
+    {
+        return Container(c.x + c.x)
+    } 
+}
+```
+We run into two problems: C# doesn't have an operator `-+-` nor can we define one. If implemented as a method, `-+-` is unpronounceable in C#. Taking the approach of using a prefix like "Op_" or "Oper_" or "Operator_" we can make it clear that this was an operator. Then the operator name would need to be run through a sanitizer to turn it into something pronounceable. The best we could hope for is getting a name out the other side like "Op_PlusMinusPlus".
```

Comment:
Perhaps a link to a C# issue? This seems like a concept that is going to get lost in this document rather than trackable. In this case linking to something on github helps avoid losing this.